app/assets/javascripts/interactive-sizing.js

```diff
@@ -1,42 +1,65 @@
-      // That will automatically handle question header (when interactive saves state) and all the possible margins
-      // and padding. Also, take into account a pinned interactive offset (space between the window top border and
-      // the interactive container top edge).
+    var isPinned = $pinned.length > 0;
+    var multipleInteractives = isPinned && $pinned.children().length > 1;
+
+    // Reset dimensions.
+    setInteractiveWidth($iframe, '100%');
+
+    if (isPinned && multipleInteractives) {
+      // This case is treated in a basic way for now. MAX setting doesn't make much sense.
+      // We just set aspect ratio (either provided by user or interactive) and we don't limit interactive height.
+      // So, interactives might get 'unpinned' as a result. It could have been treated better in the future,
+      // if there's a need for that.
+      $iframe.css('height', $iframe.width() / aspectRatio);
+      return;
+    }
+
+    if (isPinned) {
+      // If interactive is pinned (and there's only one interactive), make sure that maxHeight calculations don't
+      // cause interactive to become unpinned (as it gets too tall). Calculate the current difference between height
+      // of the interactive and its container. That will automatically handle question header (when interactive saves
+      // state) and all the possible margins and padding. Also, take into account a pinned interactive offset (space
+      // between the window top border and the interactive container top edge).
       var headersHeight = $pinned.height() - $iframe.height();
       maxHeight -= headersHeight + window.SCROLL_INTERACTIVE_MOD_OFFSET;
     }
-    $iframe.attr('width', '100%');
 
     if (resizeMethod === 'MAX') {
-      $iframe.height(maxHeight);
+      $iframe.css('height', maxHeight);
     }
     else if (resizeMethod === 'MANUAL') {
-      $iframe.height($iframe.width() / aspectRatio);
+      $iframe.css('height', $iframe.width() / aspectRatio);
     }
     else /* (resizeMethod === 'DEFAULT') */ {
-      $iframe.height($iframe.width() / aspectRatio);
-      if ($iframe.height() > maxHeight) {
-        var scale = (maxHeight / $iframe.height()) * 100 + '%';
-        $iframe.attr('width', scale);
-        $iframe.height(maxHeight);
-        // Rescale question header too (it's present when interactive saves state).
-        var $header = $iframe.parents(".embeddable-container").find(".question-hdr")
-        if ($header.length > 0) {
-          $header.css('width', scale);
-        }
-      }
+      $iframe.css('height', $iframe.width() / aspectRatio);
+      // PJ [7/16/2019]: Why is the height limited only in the DEFAULT mode, but not in MANUAL?
+      // I don't find this intuitive, but it has been implemented because of some issues described
+      // in this story: https://www.pivotaltracker.com/n/projects/736901/stories/161458506
```

Comment:
That story describes it well. For interactives that don't scale well, we need a way to force them to have a height larger than the maxHeight.  

Additionally, @dougmartin is about to add support for interactives that dynamically request more height.  So in this case we'll have interactives that need more than the maxHeight and require a page scroll to see the whole interactive. 

There is probably a better way to handle all this but we haven't figured it out yet. The variables that need to be considered are:
- not too confusing for authors
- works well when author changes page layout
- supports the variety of sizing options required by interactives and authors